File: 78-subsets/subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        int n = nums.size();
        vector<vector<int>> sol;
        vector<int> a;
        auto bkt = [&](auto&& self, int idx){
            if(idx>=n){
                sol.push_back(a);
                return;
            }
            a.push_back(nums[idx]);
            self(self,idx+1);
            a.pop_back();
            self(self,idx+1);
        };

        bkt(bkt,0);
        return sol;
    }
};

File: 78-subsets/subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        n=len(nums)
        limit = 2**n
        sol = []

        for mask in range(limit):
            subset  = []
            for idx in range(n):
                if mask & (1<<idx):
                    subset.append(nums[idx])
            sol.append(subset)
    
        return sol